frontend/catalyst/pennylane_extensions.py

```diff
@@ -378,8 +395,14 @@ def grad(f: DifferentiableLike, *, method=None, h=None, argnum=None):
 
     .. warning::
 
-        Currently, higher-order differentiation or differentiation of non-QNode functions
-        is only supported by the finite-difference method.
+        If ``method="defer"`` is specified, Catalyst supports ``diff_method="parameter-shift"``
+        and ``diff_method="adjoint"`` on internal QNodes. Notably, the default QNode
+        differentiation method of ``"finite-diff"`` is not supported when used with ``"defer"``.
```

Comment:
Ok so I spoke with Josh and we decided on the following:
- duplication of the doc strings is a necessary evil for the benefit of the user, so we need to copy all changes here to the Jacobian function as well
- the "note" box doesn't need to be all the way at the top here, it should be moved to just above the first example
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/eac440b1-cbd7-4d43-90ad-22aeb7889656)
- the short warning can stay up at the top
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/961c9333-7e24-4924-9945-6e86759a9fea)
- the longer warning doesn't need to be a warning and should be incorporated into the explanation of the `diff_method` argument
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/6a35d213-b931-4402-96f1-1e76bcb909c4)
- the name `"defer"` should be changed to `"auto"`, and should come first in this list since it is the default now
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/162fba20-6aa3-4c7e-8c0d-023381f85044)

Reply:
But we can do this in a quick follow up PR as well, from my side this is good to go 🙂